packages/mds-web-sockets/clients.ts

```diff
@@ -28,31 +33,47 @@ export class Clients {
     return key.publicKey || key.rsaPublicKey
   }
 
-  public constructor(supportedEntities: readonly string[]) {
+  public constructor(supportedEntities: SupportedEntities) {
     // Initialize subscription list with configured entities
-    this.subList = Object.fromEntries(supportedEntities.map(e => [e, []]))
+    this.subList = Object.fromEntries(Object.keys(supportedEntities).map(e => [e, []]))
     this.authenticatedClients = []
     this.saveClient = this.saveClient.bind(this)
+    this.supportedEntities = supportedEntities
+  }
+
+  public hasScopes(neededScopes: string[], client: WebSocket) {
+    const clientEntry = this.authenticatedClients.find(x => x.socket === client)
```

Comment:
Seems like this could be optimized by using a `Map` with the socket/client as the key.